README.md

```diff
@@ -4,6 +4,113 @@
 
 This library provides instrumentation for CDI modules in Java EE applications.
 
+## Usage
+
+For a concrete and comprehensive set of examples, refer to the `opentracing-cdi-example`
+module.
+
+### Injecting a tracer
+
+In your managed beans, you can get an instance of the appropriate tracer via:
+
+```java
+@Inject
+Tracer tracer;
+```
+
+If you have your own `Tracer` producer, you'll need to mark your own as an
+`@Alternative`
```

Comment:
It's weird because they are using it for test impl. Other thing is that alternative bean has to be defined in `beans.xml` Am I right? If this is true it makes it harder to configure.

```
<beans ... >
    <alternatives>
        <class>encoder.TestCoderImpl</class>
    </alternatives>
</beans>
```

Is there a way to make user defined producer as a primary? The best without any additional configuration.